src/ol/imagetile.js

```diff
@@ -60,11 +60,111 @@ ol.ImageTile = function(tileCoord, state, src, crossOrigin, tileLoadFunction) {
    */
   this.tileLoadFunction_ = tileLoadFunction;
 
+  /**
+   * @private
+   * @type {number}
+   */
+  this.sequenceNumber_ = 0;
+
+
+  /**
+   * @private
+   * @type {ol.ImageTile}
+   */
+  this.outOfBandTile_ = null;
+
 };
 goog.inherits(ol.ImageTile, ol.Tile);
 
 
 /**
+ * set the url for the out-of-band tile
+ *
+ * @param {string} url The url to load out-of-band.
+ * @return {ol.ImageTile}
+ */
+ol.ImageTile.prototype.setOutOfBandUrl = function(url) {
+  if (this.outOfBandTile_ === null) {
+
+    // create a new tile
+    this.outOfBandTile_ = this.createOutOfBandTile_(url);
+    this.assertSequence_();
+    return this.outOfBandTile_;
+
+  } else if (this.outOfBandTile_.state == ol.TileState.IDLE) {
+    goog.asserts.assert(this.outOfBandTile_.outOfBandTile_ === null);
+
+    // replace the old tile, and cause it to be dropped from the tile queue
+    var newTile = this.createOutOfBandTile_(url);
+    // this sets the tile load priority to DROP
+    this.outOfBandTile_.tileCoord = [-1, 0, 0];
+    this.outOfBandTile_ = newTile;
+    this.assertSequence_();
+    return this.outOfBandTile_;
+
+  } else if (this.outOfBandTile_.state == ol.TileState.LOADING ||
+      this.outOfBandTile_.state == ol.TileState.LOADED ||
+      this.outOfBandTile_.state == ol.TileState.ERROR) {
+    return this.outOfBandTile_.setOutOfBandUrl(url);
+  } else {
+    throw "don't know what to do when outOfBandTile is in state" +
+        this.outOfBandTile_.state;
```

Comment:
Please always throw `Error`s instead of strings.

Throughout the majority of the library, we use `goog.asserts.fail` instead of `throw` (exceptions are`wktformst.js` and `rbush.js`).  I don't have a strong opinion, but it would be more consistent to `goog.asserts.fail`.  The benefit is that error strings are not included in production code.  The drawback is that you get unexpected/silent failures in production.